platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -2356,14 +2359,20 @@ Found the following transforms:
                 }
             }
 
-            ${configurationAndTransform('NameManglingTransform')}
-        """
+            ${declareTransform('NameManglingTransform')}
 
-        when:
-        run "resolve"
+            task hasUniqueArtifactIds {
+                def artifacts = configurations.compile.incoming.artifactView {
+                    attributes { it.attribute(artifactType, 'size') }
+                }.artifacts
+                doLast {
+                    assert (artifacts.collect { it.id } as Set).size() == 2
+                }
+            }
+        """
 
-        then:
-        outputContains("ids: [out-foo.txt (test:test:1.3), out-bar.txt (test:test:1.3)]")
```

Comment:
🤔 Isn't this trying to check/assert that the two artifacts are different even though they have the same artifact id?

Reply:
Nevermind